pages/p/[id].tsx

```diff
@@ -42,37 +53,37 @@ export const getServerSideProps: GetServerSideProps = async ({ params, res}) =>
  
 };
 
-const Post: React.FC<PostProps & prismaError> = (props) => {
-  let title = props?.title
-  let name = props.code ? props.name : props?.owner?.name
+const Post: React.FC<PropObj> = (props) => {
+  let title = props?.post?.title
+  let name = props.code ? props.name : props?.post?.owner?.name
   return (
     <Layout>
       {props?.code ? <Error statusCode={404} title="Post does not exist"/> : 
-        <div>
-          <h2 data-testid="heading-title">{title}</h2>
-          <p data-testid="post-name">{name}</p>
-          <style jsx>{`
-        .page {
-          background: white;
-          padding: 2rem;
-        }
+        !props?.Session?.user ? <p>need to login</p> : <div>
+        <h2 data-testid="heading-title">{title}</h2>
+        <p data-testid="post-name">{name}</p>
+        <style jsx>{`
```

Comment:
Using a ternary within a ternary is not easy to read. Extract this logic into a function to render the components using conditional statements.